src/eva/vision/data/datasets/classification/wsi.py

```diff
@@ -1,40 +1,109 @@
-from eva.vision.data.datasets.wsi import MultiWsiDataset
+from eva.vision.data.datasets import wsi
+from eva.vision.data.datasets.classification import base
+from eva.vision.data.wsi.patching import samplers
 
 
-class MultiWsiClassificationDataset(MultiWsiDataset):
-    """Classification Dataset class for reading patches from multiple whole-slide images.
+class WsiClassificationDataset(wsi.MultiWsiDataset, base.ImageClassification):
+    """A general dataset class for whole-slide image classification using manifest files."""
 
-    # TODO: Replace this by dataset specific classes?
-    """
+    default_column_mapping: Dict[str, str] = {
+        "path": "path",
+        "target": "target",
+        "split": "split",
+    }
+
+    def __init__(
+        self,
+        root: str,
+        manifest_file: str,
+        width: int,
+        height: int,
+        target_mpp: float,
+        sampler: samplers.Sampler,
+        backend: str = "openslide",
+        split: Literal["train", "val", "test"] | None = None,
+        image_transforms: Callable | None = None,
+        column_mapping: Dict[str, str] = default_column_mapping,
+    ):
+        """Initializes the dataset.
+
+        Args:
+            root: Root directory of the dataset.
+            manifest_file: The path to the manifest file, relative to
+                the `root` argument. The `path` column is expected to contain
+                relative paths to the whole-slide images.
+            width: Width of the patches to be extracted, in pixels.
+            height: Height of the patches to be extracted, in pixels.
+            target_mpp: Target microns per pixel (mpp) for the patches.
+            sampler: The sampler to use for sampling patch coordinates.
+            backend: The backend to use for reading the whole-slide images.
+            split: The split of the dataset to load.
+            image_transforms: Transforms to apply to the extracted image patches.
+            column_mapping: Mapping of the columns in the manifest file.
+        """
+        self._root = root
+        self._manifest_file = manifest_file
+        self._split = split
+        self._width = width
+        self._height = height
+        self._target_mpp = target_mpp
+        self._sampler = sampler
+        self._backend = backend
+        self._image_transforms = image_transforms
+        self._column_mapping = self.default_column_mapping | column_mapping
+
+        self._manifest = self._load_manifest(os.path.join(self._root, self._manifest_file))
+        self._file_paths = self._manifest[self._column_mapping["path"]].tolist()
+
+        wsi.MultiWsiDataset.__init__(
+            self,
+            root=root,
+            file_paths=self._file_paths,
+            width=width,
+            height=height,
+            sampler=sampler,
+            target_mpp=target_mpp,
+            backend=backend,
+            image_transforms=image_transforms,
+        )
```

Comment:
maybe we should call super at the start and then register (self._*) the attributes that are not in the parent class; like I guess `self._root` should be available in the parent class right?

Reply:
Great catch! I updated this now (I had to register some attributes before the super call though, because I have to load the manifest / list of file paths first)